packages/grafana-runtime/src/services/pluginExtensions/getPluginExtensions.ts

```diff
@@ -28,3 +30,19 @@ function getPluginExtensionGetter(): GetPluginExtensions {
 }
 
 export const getPluginExtensions: GetPluginExtensions = (options) => getPluginExtensionGetter()(options);
+
+export const getPluginLinkExtensions: GetPluginExtensions<PluginExtensionLink> = (options) => {
+  const { extensions } = getPluginExtensions(options);
+
+  return {
+    extensions: extensions.filter(isPluginExtensionLink),
+  };
+};
+
+export const getPluginComponentExtensions: GetPluginExtensions<PluginExtensionComponent> = (options) => {
+  const { extensions } = getPluginExtensions(options);
+
+  return {
+    extensions: extensions.filter(isPluginExtensionComponent),
+  };
+};
```

Comment:
What do you think about adding a filter function to options instead?

Reply:
I'm not sure which one I prefer but the only pro with having a filter function in options is that you can filter on other dimensions than the extension type.

Reply:
Then I would prefer keeping the ones you already have in the PR. I was thinking on something like:

```ts
getPluginExtensions({
    pluginExtensionId: '',
    filter: isPluginExtensionComponent,
});
```

In that case you could write your own function if you, for example, would like to filter on plugin Id or some other property.

Reply:
(I am resolving this thread to make the PR easier to scan, and as I think we decided on keeping the current ones for now - let me know if you disagree and then we can "unresolve" it.)